### ISSUE TITLE
I am going to download yolov8.yaml and rename as  yolov8n.yaml and  then it will automatically call   yolov8n.yaml model am i right?

### ISSUE THREAD

**Issue by TapendraBaduwal** [ultralytics/ultralytics#2319]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question
I am going to download yolov8.yaml and rename as  yolov8n.yaml and  then it will automatically call   yolov8n.yaml model am i right?

**Comment by glenn-jocher** [comment]
@TapendraBaduwal thank you for reaching out. No, that is not correct. Renaming the `yolov8.yaml` file to `yolov8n.yaml` will not make the model automatically call the `yolov8n` model. In fact, if you rename the `yolov8.yaml` file, the code will not find the model configuration file and therefore will not be able to run the model. If you would like to modify the model configuration, please make sure to modify the existing `yolov8.yaml` file and save it with the same name.

**Comment by TapendraBaduwal** [comment]
@glenn-jocher I want to  use yolov8n.yaml file in my local vscode  by giving path on below code how can i do that ....How can i modify some training parameters such as learning rate (lr), batch size (batch-size), number of epochs (epochs) from my local machine .
model = YOLO("path/yolov8n.yaml")

**Comment by glenn-jocher** [comment]
@TapendraBaduwal thank you for reaching out! If you would like to provide the path to your own `yolov8n.yaml` file, you need to first save a copy of the `yolov8.yaml` file to your local machine and modify the contents according to your needs. Once you are satisfied with the changes, save the file with the name `yolov8n.yaml` in the same directory as your Python code. Then, you can update the `path` variable in the `model = YOLO('path/yolov8n.yaml')` line to reflect the path where your `yolov8n.yaml` file is saved. Note that you can replace `path` with the specific file path on your machine where the file is saved.

**Comment by Elizbellou** [comment]
@glenn-jocher hi there, regarding this issue, I want to add a few layers in the backbone by editing yolov8.yaml. Should then save it with different name or replace the existing leaving the name as it is? And what would be the argument to pass it for training? Model? Please note that I want to use pretrained model yolov8s.pt, not train from scratch. Is it possible? Thank u in advance for ur answer

**Comment by glenn-jocher** [comment]
@Elizbellou hello! Thank you for reaching out. If you want to add layers to the yolov8 backbone, you should save the modified `yolov8.yaml` with a different name (i.e. not overwrite the original `yolov8.yaml` file). Then, when you are training your model, you will pass in the path to the modified configuration file as an argument to the `model = YOLO()` function call. For example, if your modified configuration file is saved as `yolov8_extra_layers.yaml` in the same directory as your Python code and pretrained model, you would initialize the model by calling `model = YOLO('yolov8_extra_layers.yaml', 'yolov8s.pt')`, where `yolov8s.pt` is the path to the pretrained model. This will load the modified configuration file and the pretrained weights. I hope this helps! Let me know if you have any further questions.

**Comment by Elizbellou** [comment]
Thank you for your answer. I use CLI, but unfortunately when I call model=
"path-to-extralayer.yaml", " path-to-yolov8s.pt" , only one of them is
recognized when training initiates. When custom yaml file is passed it
starts training from scratch. Should I define the pretrained weights in the
config file or modify another file, e.g. train.py?

Thank you,
Elizabeth Bellou

Στις Τετ, 3 Μαΐ 2023, 05:35 ο χρήστης Glenn Jocher ***@***.***>
έγραψε:
